src/Delegate.php

```diff
@@ -25,12 +25,12 @@ class Delegate implements DelegateInterface
 
     /**
      * @param array $middleware
-     * @param callable $default
+     * @param DelegateInterface $lastDelegate
      */
-    public function __construct(array $middleware, callable $default)
+    public function __construct(array $middleware, DelegateInterface $lastDelegate)
```

Comment:
This is not correct. The default should not be a delegate, because this is perfectly acceptable for `$default`:

```php
$default = [$responseFactory, 'createResponse'];
```